### PR TITLE
Implement node

### PR DESCRIPTION
**Pull request by shellwedance**
DONE
- Implement HostSpec class
- Implement Node class
- Move ssh util function to Node method

TODO
- Implement package installation method of Node and its unit test
- Enhance unit tests to mitigate corner cases

### CODE REVIEW

**Review by SangHyeok-Kim95 — COMMENTED** [review]
몇 가지 수정이 필요한 부분이 있는 것 같습니다. 확인 부탁 드립니다.

File: pkg/node/host_spec.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
+package node
+
+import (
+	commonWrapper "hypersds-provisioner/pkg/common/wrapper"
```
[the remark this answers is not in the document]
Comment (shellwedance) [reply pkg/node/host_spec.go:4]: node와 config 등 모두 wrapper가 있으므로 일반 wrapper로 쓰면 혼동이 올 수 있을 것 같고, 그러면 `common`으로 바꾸겠습니다

File: pkg/node/wrapper.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package node
```
[the remark this answers is not in the document]
Comment (shellwedance) [reply pkg/node/wrapper.go]: wrapper에는 wrapping 관련 내용만 추가하는 것 괜찮은 것 같습니다. 좀 더 정확한 표현을 위해 `wrapper.go`를 `new_node_wrapper.go`로 수정하였고, `node.go`로 struct와 function을 이동하였습니다.

File: pkg/node/host_spec.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (SangHyeok-Kim95) [reply pkg/node/host_spec.go]: 의견 반영 감사합니다.

File: pkg/node/wrapper.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (SangHyeok-Kim95) [reply pkg/node/wrapper.go]: 반영된 내용 확인했습니다.

[the remark this answers is not in the document]
Comment (shellwedance) [reply pkg/node/wrapper.go]: node_wrapper.go로 이름 수정하였습니다.

**Review by SangHyeok-Kim95 — APPROVED** [review]
피드백 반영 감사합니다.

**Review by hookak — APPROVED** [review]
lgtm
